### ISSUE TITLE
Support optional UA (for use with task runners, eg grunt)

### ISSUE THREAD

**Issue by monolithed** [polyfillpolyfill/polyfill-service#253]
I'd like to use this service with Grunt like: 

``` js
'polyfill-service': {
    options: {
        features: [
            {
                name : 'Element.prototype.matches',
                flags: ['gated']
            },
            {
                name: 'modernizr:es5array'
            }
        ]
    },

    dest: 'polyfills.js'
}
```

What are some suggestions?

**Comment by triblondon** [comment]
Hi @monolithed, this is an interesting idea, so you're saying you want to use the polyfill service as a grunt task?  And you want to do this without specifying a UA.  This is a similar problem conceptually to #252, in that:
- We would need to serve all the requested polyfills, since we don't know which ones are actually needed
- For polyfills with browser specific variants, we would have to serve the default variant (which may be blank if there's no default variant for a polyfill)

I am concerned about the effects of this on very legacy UAs or low powered devices that may fail badly at parsing the large polyfill bundle.  For example I have a fairly new LG television with a built in web browser, which is some very old version of webkit.  If I load my employer's website, which contains a lot of script, the TV actually turns itself off.  It would be better to not serve script to UAs that can't handle it.  I remember similar crash effects happening on older BlackBerry devices prior to v6.

**Comment by monolithed** [comment]
@triblondon, 

A possible solution to the first problem — a central polyfill archive / repository. 
The second problem can be solved by the [Autopolyfiller](https://github.com/azproduction/autopolyfiller)

However, to solve the second problem to be solved first.
It is for this reason the Autopolyfiller still uses @jonathantneal's [polyfill](https://github.com/jonathantneal/polyfill)

And I'd like to draw your attention to the following [issue/idea](https://github.com/azproduction/autopolyfiller/issues/15)

**Comment by monolithed** [comment]
Perhaps someone will be interesting my [grunt-task](https://www.npmjs.com/package/grunt-polyfill-service) :yum:

**Comment by triblondon** [comment]
In #252, I have suggested an `enableUnknown` property.  Your grunt task could embed that value so that when used via grunt, the service provides a polyfill set for the generic UA case, while users of the CDN will still get an empty file for an unknown UA, unless they explicitly opt in.  Would that work?

**Comment by monolithed** [comment]
I like, but do you mean [unknown](https://github.com/Financial-Times/polyfill-service/pull/276) option?

**Comment by triblondon** [comment]
Yeah, I changed it to `unknown` following a suggestion from @matthew-andrews

**Comment by triblondon** [comment]
`unknown` has shipped in 1.2, so considering this closed.
